src/Bicep.Core/Semantics/Namespaces/AzNamespaceType.cs

```diff
@@ -489,6 +489,38 @@ static IEnumerable<FunctionOverload> GetBicepFilePermittedOverloads()
                     .WithRequiredParameter("resourceType", LanguageConstants.String, "The type of resource within the specified namespace")
                     .Build();
 
+                yield return new FunctionOverloadBuilder("toLogicalZone")
+                    .WithReturnType(LanguageConstants.String)
+                    .WithGenericDescription("Returns the logical zone corresponding to the given physical zone.")
+                    .WithRequiredParameter("subscriptionId", LanguageConstants.String, "The subscription ID of the deployed availability zones")
+                    .WithRequiredParameter("location", LanguageConstants.String, "The location of the availability zone mappings")
+                    .WithRequiredParameter("physicalZone", LanguageConstants.String, "The physical zone to convert")
+                    .Build();
+
+                yield return new FunctionOverloadBuilder("toLogicalZones")
+                    .WithReturnType(new TypedArrayType(LanguageConstants.String, TypeSymbolValidationFlags.Default, minLength: 1))
```

Comment:
>minLength: 1) [](http://example.com/codeflow?start=116&length=13)

For the plural functions, do we allow empty zones array in the input? (If yes, then the function would return an empty array.)